### ISSUE TITLE
Describe block not showing in console output with nested Context blocks

### ISSUE THREAD

**Issue by camusicjunkie** [pester/Pester#1688]
## 1. General summary of the issue

If a Describe block has nested Context blocks then the console output does not show the Describe block. This is with `-Output Detailed`. 

## 2. Describe Your Environment

Pester version     : 5.1.0 C:\Program Files\WindowsPowerShell\Modules\Pester\5.1.0\Pester.psd1
PowerShell version : 5.1.19041.1
OS version         : Microsoft Windows NT 10.0.19041.0

## 3. Expected Behavior

The Describe block should show in the console output with `-Output Detailed`

## 4.Current Behavior

When a Context block is nested within a Describe, the Describe no longer shows in the console output.
```powershell
BeforeDiscovery {
     New-Item $PSScriptRoot\test1.xlsm -Force
     New-Item $PSScriptRoot\test2.xlsm -Force

     $files = Get-ChildItem $PSScriptRoot\*.xlsm
}

Describe 'Test <_.Name>' -ForEach $files {
    Context 'Files' {
        It 'does something' {
            $true | Should -BeTrue
        }
    }
}
```
```powershell
Starting discovery in 1 files.
Discovering in E:\Module\Tests\Files.Tests.ps1.
Found 2 tests. 353ms
Discovery finished in 483ms.

Running tests from 'E:\Module\Tests\Files.Tests.ps1'
 Context Files
   [+] does something 143ms (26ms|117ms)
 Context Files
   [+] does something 33ms (3ms|30ms)
Tests completed in 1.42s
Tests Passed: 2, Failed: 0, Skipped: 0 NotRun: 0
```

**Comment by fflaten** [comment]
Seems to be related with the postponed writing of blocks to the output. I get lost in all the callbacks, but think this here might be a key part of the trouble. Pester forgets describe because it doesn't include any tests as a direct child.
https://github.com/pester/Pester/blob/836565b1586a7008e9ba690003837ae9d6b6773a/src/functions/Output.ps1#L662-L669

This also results in a weird output when you include both context and test directly under describe.

```
Describe 'Test describe context' {
    Context 'Files' {
        It 'does something' {
            $true | Should -BeTrue
        }
    }
    It 'does something' {
        $true | Should -BeTrue
    }
 }
```

```
Starting discovery in 1 files.
Discovering in /workspaces/Pester/demoDescribeForeach.tests.ps1.
Found 2 tests. 43ms
Discovery finished in 47ms.

Running tests from '/workspaces/Pester/demoDescribeForeach.tests.ps1'
 Context Files
   [+] does something 4ms (2ms|2ms)
Describing Test describe context
  [+] does something 4ms (2ms|2ms)
Tests completed in 92ms
Tests Passed: 2, Failed: 0, Skipped: 0 NotRun: 0
```

Works as intended In Pester 5.0.5
```
Running tests from '/workspaces/Pester/demoDescribeForeach.tests.ps1'
Describing Test describe context
 Context Files
   [+] does something 18ms (10ms|8ms)
  [+] does something 14ms (7ms|7ms)
Tests completed in 249ms
Tests Passed: 2, Failed: 0, Skipped: 0 NotRun: 0
```

**Comment by nohwnd** [comment]
@fflaten yup you are right, I mentioned the cause on slack, but unfortunately it did not make it into the issue. Should have come here and make sure it is in the issue.

The cleanest way would probably be adding more callbacks to the plugin. There is one that runs before running BeforeAll, but there is none that runs after BeforeAll.

**Comment by fflaten** [comment]
Solved by #1717
